File: CS/BlazorClientApp/Services/DemoLocalizationService.cs
using DevExpress.Blazor.Localization;
using System;
using System.Collections.Generic;
using System.Globalization;
using System.Linq;
using System.Resources;
using System.Threading.Tasks;

namespace BlazorClientApp.Services
{
    public class DemoLocalizationService : DxLocalizationService, IDxLocalizationService {
        ResourceManager _resourceManager;
        ResourceManager ResourceManager {
            get {
                if(_resourceManager == null)
                    _resourceManager = new ResourceManager("BlazorClientApp.Resources.LocalizationRes", typeof(DemoLocalizationService).Assembly);
                return _resourceManager;
            }
        }
        string IDxLocalizationService.GetString(string key) {
            var culture = CultureInfo.CurrentUICulture.Name;
            string value = null;
            switch(culture) {
                case "it-IT":
                    value = ResourceManager.GetString(key);
                    break;
            }
            return value ?? base.GetString(key);
        }
    }
}

File: VB/BlazorClientApp/Services/DemoLocalizationService.vb
Imports DevExpress.Blazor.Localization
Imports System.Globalization
Imports System.Resources

Namespace BlazorClientApp.Services

    Public Class DemoLocalizationService
        Inherits DxLocalizationService
        Implements IDxLocalizationService

        Private _resourceManager As ResourceManager

        Private ReadOnly Property ResourceManager As ResourceManager
            Get
                If _resourceManager Is Nothing Then _resourceManager = New ResourceManager("BlazorClientApp.Resources.LocalizationRes", GetType(DemoLocalizationService).Assembly)
                Return _resourceManager
            End Get
        End Property

        Private Overloads Function IDxLocalizationService_GetString(ByVal key As String) As String Implements IDxLocalizationService.GetString
            Dim culture = CultureInfo.CurrentUICulture.Name
            Dim value As String = Nothing
            Select Case culture
                Case "it-IT"
                    value = ResourceManager.GetString(key)
            End Select

            Return If(value, GetString(key))
        End Function
    End Class
End Namespace
